josh-core/src/filter/mod.rs

```diff
@@ -824,12 +824,18 @@ pub fn unapply<'a>(
     }
 
     if let Op::Chain(a, b) = to_op(filter) {
-        let i = if let Ok(i) = invert(a) { invert(i)? } else { a };
-        let p = apply(transaction, i, parent_tree.clone())?;
+        // If filter "a" is invertable, use "invert(invert(a))" version of it, otherwise use as is
```

Comment:
This comment states the obvious but does not explain why the double invert is done. So it's a good this that you poke at this because it makes me realize that I don't remember why this is done and should have commented back when I wrote it... I'll investigate.